gitea/docker-compose.yml

```diff
@@ -55,7 +55,7 @@ services:
     environment:
       - GITEA_INSTANCE_INSECURE=1
       - GITEA_RUNNER_REGISTRATION_TOKEN=eMdEwIzSo87nBh0UFWZlbp308j6TNWr3WhWxQqIc
-      - GITEA_INSTANCE_URL=http://server:3000
+      - GITEA_INSTANCE_URL=http://host.docker.internal:3000
```

Comment:
For...reasons...we use `gitea.local` rather than `host.docker.internal`. I forget for the moment exactly those reasons. Possibly because we want the hostname to be usable from the host as well as inside containers. Possibly also we want to be able to target a gitea that is running somewhere other than on the host?

Reply:
The GITEA_INSTANCE_URL  is used by act_runner for two things:  (1) what endpoint to poll for new tasks,  (2) for constructing the URL to pass to any spawned task executors for checking out code.

Unfortunately neither `server` (the container name in the stack) nor `gitea.local` are resolvable for both (1) and (2).  `server` works for (1) but not (2) (since spawned containers are not in the same stack) and gitea.local won't work for either, since it isn't used _inside_ docker.

The only name thing I've found that works universally is to reach back to the exposed port on the host.

Reply:
(Note, if we could somehow make sure that any newly spawned task containers were attached to the same stack, we could likely use `server`.)

Reply:
Hmm. I thought `host.docker.internal` wasn't defined in all Docker installs?